nameko/extensions.py

```diff
@@ -72,19 +73,26 @@ def kill(self):
         swallow the exception to allow the container kill to continue.
         """
 
-    def clone(self, container):
-        if self.is_clone:
-            raise RuntimeError('Cloned extensions cannot be cloned.')
+    def bind(self, container):
+        """ Get an instance of this Extension to bind to `container`.
+        """
+
+        def clone(prototype):
+            if prototype.is_clone:
```

Comment:
Yep, ``is_clone`` will become ``is_bound``. Cloning is an implementation detail.